homeassistant/components/unifiled/light.py

```diff
@@ -0,0 +1,106 @@
+"""Support for Unifi Led lights."""
+import logging
+
+import voluptuous as vol
+
+from unifiled import unifiled
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    PLATFORM_SCHEMA,
+    SUPPORT_BRIGHTNESS,
+    Light,
+)
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_PORT, CONF_USERNAME
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+# Validation of the user's configuration
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Optional(CONF_PORT, default="20443"): cv.string,
+    }
+)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Unifi LED platform."""
+
+    # Assign configuration variables.
+    # The configuration check takes care they are present.
+    _ip = config[CONF_HOST]
+    _port = config[CONF_PORT]
+    _username = config[CONF_USERNAME]
+    _password = config[CONF_PASSWORD]
+
+    api = unifiled(_ip, _port, username=_username, password=_password)
+
+    # Verify that passed in configuration works
+    if not api.getloginstate():
+        _LOGGER.error("Could not connect to unifiled controller")
+        return
+
+    # Add devices
+    add_entities(UnifiLedLight(light, api) for light in api.getlights())
+
+
+class UnifiLedLight(Light):
+    """Representation of an unifiled Light."""
+
+    def __init__(self, light, api):
+        """Init Unifi LED Light."""
+
+        self._api = api
+        self._light = light
+        self._name = light["name"]
+        self._unique_id = light["id"]
+        self._state = light["status"]["output"]
+        self._brightness = self._api.convertfrom100to255(light["status"]["led"])
```

Comment:
IIRC doing I/O in `__init__` is unwanted, so these should be done in `update()`? That is, if the I/O is not done already in the setup..

Reply:
The io is done in the setup. convertfrom100to255() only does a bit of math to convert to a range ha likes.